modules/javafx.graphics/src/main/java/com/sun/javafx/font/MacFontFinder.java

```diff
@@ -72,6 +77,13 @@ public static boolean populateFontFileNameMap(
         if (locale == null) {
             locale = Locale.ENGLISH;
         }
+        try {
+            Stream<Path> stream = Files.list(Paths.get("/System/Library/Fonts"));
+            stream.forEach(f -> PrismFontFactory.getFontFactory().registerEmbeddedFont(f.toString()));
+        } catch (IOException e) {
```

Comment:
registerEmbeddedFont() is intended to be used for fonts embedded in an application not for system fonts. 
Also is there anywhere else we are hard-coding /System/Library/Fonts. Apple move things around.
Also as well as likely needing to use a different call to register, you should make sure that the API
is registering all the fonts in a collection.
Finally (I think) what happens if you print ? The font info needs to be passed over to J2D.

Reply:
Another thought .. after doing this do these "." fonts appear in the list of fonts reported by
javafx.scene.font.getFontNames() ?

I suspect they really should not ..

Reply:
They are in there. But with the current approach for CT glyph-parsing, I see no other way. The parsing is done in native code (e.g. OS.CTLineCreateWithAttributedString()) but we extract the required font from the returned runs, e.g. 
`String fontName = CTFontCopyAttributeDisplayName(actualFont);`
Next, the `fontName` is searched for, but if it is not in the "public" list of fonts, this fails. So if we don't make these dot fonts public, this approach doesn't work.

Reply:
When you say "public" you mean in the lists that the implementation searches for a match ?
This could get tricky. I don't have a complete answer off the top of my head.
Could we at least filter the returned list of full and family names to exclude them ?
Then explicit searches would work but no one will be seeing them in a list of fonts available to apps.

Reply:
the filter would be macOS only and exlude fonts with a name beginning with "."

Reply:
That is possible yes. An alternative is that we don't store the {name/file} info for the fonts that we explicitly load in the same map as the one created by `MacFontFinder.populateFontFileNameMap` but in a separate Map, that is macOS-specific and only used when a FontResource is requested by a call that can be considered originating from a native request (since those should have access to the internal fonts).